### PR TITLE
Update to dotnet 8

### PR DESCRIPTION
**Pull request by Meir017**
I believe we should be able to 
- Create a dotnet tool from the `HtmlGenerator` project.
- Migrate the `SymbolsController` to pure js code (or maybe webassembly) and then integrate with docfx to embed the navigateable sourcecode into a documentation site

see passing tests - https://github.com/Meir017/SourceBrowser/pull/1

### CODE REVIEW

File: src/HtmlGenerator/HtmlGenerator.csproj [hunk 1]
```diff
@@ -52,11 +52,10 @@
     <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic.Features" Version="$(NuGetVersionRoslyn)" />
     <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic.Workspaces" Version="$(NuGetVersionRoslyn)" />
     <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.MSBuild" Version="$(NuGetVersionRoslyn)" />
-    <PackageReference Include="MSBuild.StructuredLogger" Version="2.2.169" />
-    <PackageReference Include="GuiLabs.Language.Xml" Version="1.2.88" />
-    <PackageReference Include="Microsoft.VisualStudio.Language.Intellisense" Version="16.10.230" />
```
Comment (Meir017) [inline src/HtmlGenerator/HtmlGenerator.csproj:57]: this was the only dependency requiring us to target net472, we were only using two enums from this libraries, I just copied these enums into `SymbolIdService`

**Comment by KirillOsenkov** [comment]
Hi, I apologize, but I will not be taking this as the repo is in low maintenance mode and I will only take PRs when absolutely necessary. Feel free to start your own fork.

 I don't see how you can get rid of the server side given the volume of the data. If it's possible it's definitely outside the scope of this project. Sorry!

**Comment by Meir017** [comment]
@KirillOsenkov I still didn't go deep enough into the project, but I was thinking that in a similar way to the vscode web version that downloads an index (a tar.gz file) of the repository being viewed through it's service-worker. (I just tested https://github.dev/dotnet/runtime and the downloaded index is 89 MB)

How big is the index we create here compared to vscode's index?

**Comment by KirillOsenkov** [comment]
ah, I misunderstood

You could indeed download the symbol list locally and then you don't need the server, it should be just a few MB. Still a non-trivial amount of work (and I most certainly won't have time for it)

**Pull request closed without merging**